src/glue/jobs/run_great_expectations_on_parquet.py

```diff
@@ -0,0 +1,408 @@
+        namespace (str): name of the namespace
+        key_prefix (str): s3 key prefix to save the
+            validation results to
+
+    Returns:
+        EphemeralDataContext: data context object with validation stores'
+            configuration added
+    """
+    # Programmatically configure the validation result store and
+    # DataDocs to use S3
+    context.add_store(
+        "validation_result_store",
+        {
+            "class_name": "ValidationsStore",
+            "store_backend": {
+                "class_name": "TupleS3StoreBackend",
+                "bucket": s3_bucket,
+                "prefix": f"{namespace}/{key_prefix}",
+            },
+        },
+    )
+    return context
+
+
+def add_data_docs_sites(
+    context: "EphemeralDataContext",
+    s3_bucket: str,
+    namespace: str,
+    key_prefix: str,
+) -> "EphemeralDataContext":
+    """Adds the data docs sites configuration to the context object
+        so data docs can be saved to a s3 location. This is a special
+        workaround to add the data docs because we're using EphemeralDataContext
+        context objects and they don't store to memory.
+
+    Args:
+        context (EphemeralDataContext): data context to add to
+        s3_bucket (str): name of the s3 bucket to save gx docs to
+        namespace (str): name of the namespace
+        key_prefix (str): s3 key prefix to save the
+            gx docs to
+
+    Returns:
+        EphemeralDataContext: data context object with data docs sites'
+            configuration added
+    """
+    data_context_config = DataContextConfig()
+    data_context_config["data_docs_sites"] = {
+        "s3_site": {
+            "class_name": "SiteBuilder",
+            "store_backend": {
+                "class_name": "TupleS3StoreBackend",
+                "bucket": s3_bucket,
+                "prefix": f"{namespace}/{key_prefix}",
+            },
+            "site_index_builder": {"class_name": "DefaultSiteIndexBuilder"},
+        }
+    }
+    context._project_config["data_docs_sites"] = data_context_config["data_docs_sites"]
+    return context
+
+
+def get_spark_df(
+    glue_context: GlueContext, parquet_bucket: str, namespace: str, data_type: str
+) -> "pyspark.sql.dataframe.DataFrame":
+    """Reads in the parquet dataset as a Dynamic Frame and converts it
+        to a spark dataframe
+
+    Args:
+        glue_context (GlueContext): the aws glue context object
+        parquet_bucket (str): the name of the bucket holding parquet files
+        namespace (str): the namespace
+        data_type (str): the data type name
+
+    Returns:
+        pyspark.sql.dataframe.DataFrame: spark dataframe of the read in parquet dataset
+    """
+    s3_parquet_path = f"s3://{parquet_bucket}/{namespace}/parquet/dataset_{data_type}/"
+    dynamic_frame = glue_context.create_dynamic_frame_from_options(
+        connection_type="s3",
+        connection_options={"paths": [s3_parquet_path]},
+        format="parquet",
+    )
```

Comment:
Any reason you didn't `create_dynamic_frame_from_catalog`?

Reply:
No particular reason. I see now that the `create_dynamic_frame.from_catalog` will have the benefit of not dealing with schema inferencing. Were there any pitfalls you see with using `create_dynamic_frame.from_options` over `create_dynamic_frame.from_catalog`?

Reply:
I tend to use each command in different scenarios (`from_catalog` in JSON to Parquet and `from_options` in pyspark when I don't have a Glue table over the data already). Now that I've typed that, I immediately see that you want to use `from_options` since we don't have Glue tables over our Parquet.